joint_state_publisher/package.xml

```diff
@@ -15,9 +15,9 @@
   <url type="bugtracker">https://github.com/ros/joint_state_publisher/issues</url>
 
   <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend>python-catkin-pkg</buildtool_depend>
```

Comment:
Why do we need a `buildtool_depend` on this?

Reply:
As I understand it, we need this dependency since we do:

```
from catkin_pkg.python_setup import generate_distutils_setup
```

In the setup.py.

This comment led me to realize that I had left it out of the package.xml for jsp_gui, which I've rectified in 61dcd3c

Reply:
I haven't seen that before, in the [catkin package.xml wiki](http://wiki.ros.org/catkin/package.xml) or in the [rospy makefile tutorial](http://wiki.ros.org/rospy_tutorials/Tutorials/Makefile).

It looks like `python-catkin-pkg` is a [dependency of `catkin`](https://github.com/ros/catkin/blob/kinetic-devel/package.xml), so we don't need to explicitly depend on it.

A quick github search for `python-catkin-pkg` in `package.xml` files turns up [a ton of `catkin`'s package.xml](https://github.com/search?utf8=%E2%9C%93&q=python-catkin-pkg+filename%3Apackage.xml&type=Code&ref=advsearch) in people's build spaces, but when I [filtered out `catkin`](https://github.com/search?utf8=%E2%9C%93&q=python-catkin-pkg+NOT+infrastructure+filename%3Apackage.xml&type=Code&ref=advsearch), the only packages I recognized were [`rosjava/genjava`](https://github.com/rosjava/genjava/blob/kinetic/package.xml) and [`rocon_tools/rocon_python_utils`](https://github.com/robotics-in-concert/rocon_tools/blob/devel/rocon_python_utils/package.xml). Certainly doesn't seem to be a common pattern to explicitly depend on it.

Reply:
I'm certainly not a catkin expert.  But at least one of the packages I maintain uses this pattern: https://github.com/ros/kdl_parser/blob/748357e6913408679afbb8bf1adf7d81090aaa76/kdl_parser_py/package.xml#L27 .  (The [urdf_parser_py](https://github.com/ros/urdf_parser_py) package also used to use this pattern, though that was removed in Melodic for reasons of which I'm uncertain)

In the end, I don't feel super strongly about it, though I usually like to explicitly include my direct dependencies.  That way if one of my other dependencies removes a dependency, I don't need to necessarily change my own code.

Maybe @sloretz can weigh in on this one.

Reply:
> It looks like python-catkin-pkg is a dependency of catkin, so we don't need to explicitly depend on it.

Usually I'd advocate for including what we use directly, but in this case I agree that it can be skipped since the import in `setup.py` feels to me like [boilerplate required by catkin](http://docs.ros.org/melodic/api/catkin/html/howto/format2/installing_python.html).

If it's kept, then it should be [`python-catkin-pkg-modules`](https://github.com/ros/rosdistro/blob/70b4b0b3a41a46bc28688efb79436e46f54072ad/rosdep/python.yaml#L856) since it is just the modules that are imported, and not the command line tool. This is important because `python-catkin-pkg-modules` and `python3-catkin-pkg-modules` are side-by-side installable, while `python-catkin-pkg` and `python3-catkin-pkg` are not.

Reply:
Interesting, must have missed that in my search. I only looked through the first 5-6 pages though, probably just needed to dig further.

Normally I would agree about explicit dependencies, but I suppose I was viewing the package as a core component of `catkin` and therefore not worth explicitly depending upon. There's also the py2 vs py3 logic which should probably be pulled over to this file if we're explicitly depending on it rather than letting the `catkin` package handle that logic.

Definitely would appreciate @sloretz's input.

Reply:
All right, so it sounds like the consensus is to remove it (particularly in light of the upcoming noetic python3 transition).  I'll go ahead and do that.